src/lib/user/actions/update-my-account-details.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  dbBegin,
+  dbCommit,
+  dbQuery,
+  dbRelease,
+  dbRollback,
+} from "@/lib/data/db-utils";
+import { UserActor } from "../user-actor";
+import { MyAccountDetailsUpdate } from "../user-models";
+import { PoolClient } from "pg";
+
+type Context = {
+  actor: UserActor;
+  update: MyAccountDetailsUpdate;
+};
+
+export async function updateMyAccountDetails(
+  actor: UserActor,
+  update: MyAccountDetailsUpdate,
+) {
+  const ctx: Context = { actor, update };
+  const { dbPool } = actor.getParams();
+  const conn = await dbPool.connect();
+  try {
+    await dbBegin(conn);
+    const resUpdate = await updateAccountFields(conn, ctx);
+    if (resUpdate !== "OK") {
+      return resUpdate;
+    }
+    await dbCommit(conn);
+    return "OK_UPDATED";
+  } finally {
+    await dbRollback(conn);
+    await dbRelease(conn);
+  }
+}
+
+async function updateAccountFields(
+  conn: PoolClient,
+  ctx: Context,
+): Promise<"OK" | "FAILURE_DB_UPDATE"> {
+  const { actor, update } = ctx;
+  const { userId, userMapper } = actor.getParams();
+  const { userName, userEmail, userPhoneNumber, userResidency } = update;
+  const { userEncryptedPii } = await userMapper.mapUserPiiToUserSecurePii({
+    userName,
+    userPhoneNumber,
+    userEmail,
+  });
+
+  const sql = `
+    UPDATE users
+    SET
+      user_residency = $2,
+      user_encrypted_pii = $3
+    WHERE 
+      user_id = $1
+    RETURNING 
+      user_residency,
+      user_encrypted_pii
```

Comment:
If we only need `res.rows.length` we can use `RETURNING 1` here so the DB doesn't need to transmit the long encrypted PII back over the network.

Reply:
Done ac89917f6805ef8e610682773b0da2f93298a9b3